sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -152,15 +153,25 @@ class OrcFileFormat
       assert(supportBatch(sparkSession, resultSchema))
     }
 
-    OrcConf.IS_SCHEMA_EVOLUTION_CASE_SENSITIVE.setBoolean(hadoopConf, sqlConf.caseSensitiveAnalysis)
-
-    val broadcastedConf =
-      sparkSession.sparkContext.broadcast(new SerializableConfiguration(hadoopConf))
     val isCaseSensitive = sparkSession.sessionState.conf.caseSensitiveAnalysis
+    OrcConf.IS_SCHEMA_EVOLUTION_CASE_SENSITIVE.setBoolean(hadoopConf, isCaseSensitive)
+    val broadcastedHadoopConf = if (options.isEmpty) {
+      Option.empty
+    } else {
+      Option(sparkSession.sparkContext.broadcast(new SerializableConfiguration(hadoopConf)))
+    }
+
     val orcFilterPushDown = sparkSession.sessionState.conf.orcFilterPushDown
+    val sparkConf = sparkSession.sparkContext.conf
 
     (file: PartitionedFile) => {
-      val conf = broadcastedConf.value.value
+      val conf = if (broadcastedHadoopConf.isDefined) {
+        broadcastedHadoopConf.get.value.value
```

Comment:
Oh, I see. you're passing the original one here. Okay I am good with this approach.